src/row.rs

```diff
@@ -0,0 +1,113 @@
+/// A macro for creating a [`Row`] using vec! syntax.
+///
+/// `row!` is similar to the [`vec!`] macro, but it returns a [`Row`] instead of a `Vec`.
+///
+/// # Examples
+///
+/// * Create a [`Row`] containing a vector of [`Cell`]s:
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::row;
+///
+/// let row = row!["hello", "world"];
+/// let row = row!["hello".red(), "world".red().bold()];
+/// ```
+///
+/// * Create a [`Row`] from a given [`Cell`] repeated some amount of times:
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::row;
+///
+/// let row = row!["hello"; 2];
+/// ```
+///
+/// * Use [`text!`], [`line!`] or [`span!`] macro inside [`row!`] macro.
+///
+/// ```rust
+/// # use ratatui::prelude::*;
+/// use ratatui_macros::{row, line, text, span};
+///
+/// let row = row![
+///     line!["hello", "world"], span!(Modifier::BOLD; "goodbye {}", "world"),
+///     text!["hello", "world"],
+/// ];
+/// ```
+///
+/// [`Row`]: crate::widgets::Row
+/// [`Cell`]: crate::widgets::Cell
+#[macro_export]
+macro_rules! row {
+    () => {
+        ratatui::widgets::Row::default()
+    };
+    ($cell:expr; $n:expr) => {
+        ratatui::widgets::Row::new(vec![ratatui::widgets::Cell::from($cell); $n])
+    };
+    ($($cell:expr),+ $(,)?) => {{
+        ratatui::widgets::Row::new(vec![
+        $(
+            ratatui::widgets::Cell::from($cell),
+        )+
+        ])
+    }};
+}
+
+#[cfg(test)]
+mod tests {
+    use ratatui::text::Text;
+    use ratatui::widgets::Cell;
+    use ratatui::widgets::Row;
+
+    #[test]
+    fn row() {
+        // literal
+        let row = row!["hello", "world"];
+        assert_eq!(
+            row,
+            Row::new(vec![Cell::from("hello"), Cell::from("world")])
+        );
+
+        // explicit use of span and line
+        let row = row![crate::line!("hello"), crate::span!["world"]];
+        assert_eq!(
+            row,
+            Row::new(vec![Cell::from("hello"), Cell::from("world")])
+        );
+
+        // vec count syntax
+        let row = row!["hello"; 2];
+        assert_eq!(
+            row,
+            Row::new(vec![Cell::from("hello"), Cell::from("hello")])
+        );
+
+        use crate::text;
+        let rows = [
+            row!["\u{f002}", "Find File", text!["ctrl+f"].right_aligned()],
+            row!["\u{f021a}", "Open recent", text!["ctrl+r"].right_aligned()],
+            row!["\u{f0493}", "Open config", text!["ctrl+k"].right_aligned()],
+        ];
```

Comment:
nit: perhaps consider avoiding the unicode here to simplify the tests - mainly because while this probably looks cool, it only does so when this actually runs, and it doesn't display this anywhere.